src/player.cpp

```diff
@@ -12295,9 +12295,12 @@ std::vector<const item *> player::all_items_with_flag( const std::string &flag )
     } );
 }
 
-bool player::has_item_with_flag( const std::string &flag ) const
+bool player::has_item_with_flag( const std::string &flag, bool need_charges ) const
 {
-    return has_item_with( [&flag]( const item & it ) {
+    return has_item_with( [&flag, &need_charges]( const item & it ) {
+        if( need_charges ) {
+            return it.has_flag( flag ) && it.type->tool->max_charges ? it.charges > 0 : it.has_flag( flag );
```

Comment:
You need to check `it.is_tool()` probably. And what function does `it.type->tool->max_charges` perform here?

Reply:
Check for `max_charges` is a hack to check if this item requires charges to work.